### PR TITLE
streamingest: add job level test for stream ingestion

### PR DESCRIPTION
**Pull request by adityamaru**
This change adds a test that exercises all the components of the stream
ingestion flow. It fixes some missing pieces that were discovered while
writing the test.

Informs: #59175

Release note: None

### CODE REVIEW

**Comment by adityamaru** [comment]
Last commit is rebased on top of #59112 and #59441.

File: pkg/sql/parser/parse_test.go
Comment (pbardea) [inline pkg/sql/parser/parse_test.go:1724]: Let's also add some tests for `RESTORE TABLE foo FROM ...` since it's supported by the grammar. Although we error on it later, I think it's worth noting in these tests that it's a grammar that we support. A comment along with the tests explaining that we'll error out later would be good too.

**Review by adityamaru — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59588)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pbardea)

___
*[pkg/ccl/streamingccl/streamingest/stream_ingestion_test.go, line 45 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/59588#-MSEGV4wXN9BPswcGuSc-r4-45:-MTH1EuVDZlHuociP3Ze:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/665b56d7177edf05b5ff85acae35e07e53ec6828/pkg/ccl/streamingccl/streamingest/stream_ingestion_test.go#L45)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

I think we can do this with an import line that looks like:
`_ "github.com/cockroachdb/cockroach/pkg/ccl/kvccl/kvtenantccl"`
</blockquote></details>

Done.
___
*[pkg/sql/parser/parse_test.go, line 1724 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/59588#-MSEGV4ryb2vicW5Arts-r4-1724:-MTH1IFw5aOmDAWVqvAT:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/665b56d7177edf05b5ff85acae35e07e53ec6828/pkg/sql/parser/parse_test.go#L1724)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

Let's also add some tests for `RESTORE TABLE foo FROM ...` since it's supported by the grammar. Although we error on it later, I think it's worth noting in these tests that it's a grammar that we support. A comment along with the tests explaining that we'll error out later would be good too.
</blockquote></details>

Done.

**Comment by adityamaru** [comment]
Going to wait for your timer change to merge before I check this in, but this should be RFAL!

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_test_utils.go
Comment (pbardea) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_test_utils.go:1]: Year needs to be updated.

**Review by pbardea — APPROVED** [review]
LGTM after rebase.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_test_utils.go
Comment (adityamaru) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_test_utils.go]: Done.

**Comment by adityamaru** [comment]
TFTR!

bors r=pbardea

**Pull request merged**
